runtime/include/wasm_stack.h

```diff
@@ -55,16 +56,18 @@ wasm_stack_init(struct wasm_stack *wasm_stack, uint64_t capacity)
 		goto err_stack_allocation_failed;
 	}
 
-	wasm_stack->low = (uint8_t *)mmap(wasm_stack->buffer + /* guard page */ PAGE_SIZE, capacity,
-	                                  PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
-	if (unlikely(wasm_stack->low == MAP_FAILED)) {
+	wasm_stack->low      = wasm_stack->buffer + /* guard page */ PAGE_SIZE;
```

Comment:
I noticed that the wasm_memory used `mprotect` instead of `mmap` to change the protection bits to `PROT_READ | PROT_WRITE` and wanted to change this to be consistent. I find it a bit easier than `mmap`, which has so many arguments with dummy values. Check out the comment at line 10 of this file to see an explanation of these pointers.